packages/st2/debian/st2api.service

```diff
@@ -1,19 +0,0 @@
-[Unit]
-Description=StackStorm service st2api
-After=network.target st2api.socket
-Requires=st2api.socket
-
-[Service]
-Type=simple
-User=st2
-Group=st2
-Environment="DAEMON_ARGS=-k eventlet -b 127.0.0.1:9101 --workers 1 --threads 1 --graceful-timeout 10 --timeout 30 --log-config /etc/st2/logging.api.gunicorn.conf --error-logfile /var/log/st2/st2api.log"
-EnvironmentFile=-/etc/default/st2api
```

Comment:
I just noticed that the generator does not include `EnvironmentFile` any more.

One good thing, is that makes the generator truly agnostic between debian vs EL.

One potential gotcha is that if anyone is relying on this, then that file will no longer be sourced. They can, however, add `EnvironmentFile` to a systemd service override file `/etc/systemd/system/st2api.conf.d/some-file-name-here.conf`. That can also be used to override anything else if desired. (The same applies to st2auth and st2stream)

Reply:
As far as I know, the `/etc/default/st2api` configuration file is how Debian based distributions allowed overriding sysv and upstart service scripts.  As you've rightly pointed out, people should use the built-in systemd override mechanism and all currently supported OS' are systemd based.  We can highlight this in the v3.8 upgrade notes.